src/explore-education-statistics-common/src/components/ChartsToPNG.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useExportButtonContext } from '@common/contexts/ExportButtonContext';
+import html2canvas, { Options as HTML2CanvasOptions } from 'html2canvas';
+import { useCallback, useState } from 'react';
+
+export type UseGenerateImage<T extends HTMLElement = HTMLDivElement> = [
+  (callback?: BlobCallback) => Promise<string | undefined>,
+  {
+    isLoading: boolean;
+  },
+];
+
+export type UseGenerateImageArgs = {
+  options?: HTML2CanvasOptions;
+  quality?: number;
+  type?: string;
+};
+
+export function useGenerateImage(
+  args?: UseGenerateImageArgs,
+): UseGenerateImage {
+  const [isLoading, setIsLoading] = useState(false);
+  const ref = useExportButtonContext();
+
+  const generateImage = useCallback(
+    // eslint-disable-next-line consistent-return
+    async (callback?: BlobCallback) => {
+      if (ref && ref.current) {
+        setIsLoading(true);
+
+        return html2canvas(ref.current, {
```

Comment:
Could this be refactored to use `await` instead of promises? I think we generally prefer async/await over promises, though admittedly it's not in the style guide so up to you. Same question for `useCurrentPng`